src/MMCPServer.cpp

```diff
@@ -0,0 +1,807 @@
+ */
+QPair<bool, QString> MMCPServer::chatName(const QString& name)
+{
+    setChatName(name);
+
+    if (!clients.isEmpty()) {
+        const QString outMsg = QString("%1%2%3")
+                                .arg(static_cast<char>(NameChange))
+                                .arg(name)
+                                .arg(static_cast<char>(End));
+
+        QListIterator<MMCPClient*> it(clients);
+        while (it.hasNext()) {
+            MMCPClient* cl = it.next();
+            cl->writeData(outMsg);
+        }
+    }
+
+    const QString infoMsg = tr("[ CHAT ]  - You are now known as %1.").arg(name);
+    mpHost->postMessage(infoMsg);
+
+    return {true, QString()};
+}
+
+/**
+ * Script command, sends side channel data to all clients
+ */
+QPair<bool, QString> MMCPServer::chatSideChannel(const QString& channel, const QString& msg)
+{
+    if (clients.isEmpty()) {
+        return {false, qsl("no connected clients")};
+    }
+
+    const QString outMsg = QString("%1[%2]%3%4")
+                            .arg(static_cast<char>(SideChannel))
+                            .arg(channel)
+                            .arg(msg)
+                            .arg(static_cast<char>(End));
+
+    QListIterator<MMCPClient*> it(clients);
+    while (it.hasNext()) {
+        MMCPClient* cl = it.next();
+        //Only send to Mudlet clients
+        if (cl->getVersion().contains("Mudlet")) {
+            cl->writeData(outMsg);
+        }
+    }
+
+    return {true, QString()};
+}
+
+
+/**
+ * Script command, Sends an unformatted chat message to everyone.
+ * Warning, this has the potential for abuse!  But whatever, the chat protocol allows it.
+ */
+QPair<bool, QString> MMCPServer::chatRaw(const QString& msg)
+{
+    if (clients.isEmpty()) {
+        return {false, qsl("no connected clients")};
+    }
+
+    const QString outMsg = QString("%1%2%3")
+                            .arg(static_cast<char>(TextEveryone))
+                            .arg(msg)
+                            .arg(static_cast<char>(End));
+
+    QListIterator<MMCPClient*> it(clients);
+    while (it.hasNext()) {
+        MMCPClient* cl = it.next();
+        cl->writeData(outMsg);
+    }
+
+    clientMessage(msg);
+
+    return {true, QString()};
+}
+
+
+/**
+ * Assigns a client to a chat group
+ */
+QPair<bool, QString> MMCPServer::chatSetGroup(const QVariant& target, const QString& group)
+{
+    MMCPClient* client = clientByNameOrId(target);
+
+    if (client != nullptr) {
+        const QString currentGroup = client->getGroup();
+        bool assigned = client->setGroup(group);
+
+        QString infoMsg;
+        if (assigned) {
+            infoMsg = tr("[ CHAT ]  - Assigned '%1' to group '%2'.")
+                    .arg(m_chatName).arg(group);
+        } else {
+            infoMsg = tr("[ CHAT ]  - Removed '%1' from group '%2'.")
+                    .arg(m_chatName).arg(currentGroup);
+        }
```

Comment:
I think you are using the `(QString) MMCPServer::m_chatName` here when it should be the `(QString) MMCPClient::m_chatName` of what `client` points to. Unless this is a real problem for current Medievia users I am rolling another PR that happens to rename those members and I'll fix this at the same time...

Reply:
Yea those should be client->chatName()